### PR TITLE
libgtop: new package

### PR DESCRIPTION
**Pull request by teaguesterling**
Adding libgtop needed for building the XFCE4 core.

Depends on: https://github.com/spack/spack/pull/44644 for later versions of glib.

### CODE REVIEW

**Comment by teaguesterling** [comment]
@spackbot fix style

**Comment by teaguesterling** [comment]
@spackbot rerun pipeline

File: var/spack/repos/builtin/packages/libgtop/package.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    homepage = "https://gitlab.gnome.org/GNOME/libgtop"
+    url = "https://download.gnome.org/sources/libgtop/2.41/libgtop-2.41.3.tar.xz"
+
+    maintainers("teaguesterling")
+
+    license("GPLv2", checked_by="teaguesterling")
+
+    # can be uncommented once glib@2.80 is added
+    # version("2.41.3", sha256="775676df958e2ea2452f7568f28b2ea581063d312773dd5c0b7624c1b9b2da8c")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/libgtop/package.py]: This is still missing, we should either remove it, or add the relevant versions of glib in another PR (I see the one mentioned in the description has been closed)

File: var/spack/repos/builtin/packages/libgtop/package.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    version("2.41.2", sha256="d9026cd8a48d27cdffd332f8d60a92764b56424e522c420cd13a01f40daf92c3")
+    version("2.41.1", sha256="43ea9ad13f7caf98303e64172b191be9b96bab340b019deeec72251ee140fe3b")
+
+    depends_on("pkgconfig", type="build")
+    with default_args(type=("build", "link", "run")):
+        # depends_on("glib@2.80.2:", when="@2.41.3")  # can be uncommented once glib@2.80 is added
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/libgtop/package.py]: Same

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/libgtop/package.py (hunk 1 above)
Comment (teaguesterling) [reply var/spack/repos/builtin/packages/libgtop/package.py]: There's some unresolved issues with glib 2.80 that were pointed out to me with the other PR. I kept these in comments to make additions easier in the future. 

See #44323.

**Comment by alalazo** [comment]
Let's just remove the comments, merge this, then submit another PR adding v2.41.3 Comments like these are easy to forget if a PR takes longer than expected, and then they become outdated.

**Comment by teaguesterling** [comment]
> Let's just remove the comments, merge this, then submit another PR adding v2.41.3 Comments like these are easy to forget if a PR takes longer than expected, and then they become outdated.

Done!

**Review by alalazo — APPROVED** [review]

**Pull request merged**